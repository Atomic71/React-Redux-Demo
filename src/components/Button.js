import React from 'react';
import PropTypes from 'prop-types'

// button types

// primary - color: primary 
// danger - color: red
// processing - color: grey
// dark - color: dark
//


export const Button = ({classes, clickHandler, text, type, disabled}) => (
    <button className={`button ${classes}`} onClick={clickHandler} type={type} disabled={disabled}>
        {text}
    </button>
)

Button.propTypes = {
    classes: PropTypes.string,
    clickHandler: PropTypes.func,
    text: PropTypes.string.isRequired,
    type: PropTypes.string,
    disabled: PropTypes.bool,
}