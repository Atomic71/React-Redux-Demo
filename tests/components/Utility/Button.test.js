import React from 'react';
import { Button } from '../../../src/components/Utility/Button';
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer';

describe('Button utility component', () => {
    const onClickFunc = jest.fn();
    const props = {
        type: "something",
        text: "Andrej's Button",
        onClick: onClickFunc
    }
    it('matches the expected snapshot', () => {
        const myButtonn = renderer.create(<Button {...props} />).toJSON(); 
        expect(myButtonn).toMatchSnapshot();
    });
    
    it('correctly calls the onClick function', () => {
        const myButton = shallow( <Button {...props} />
        ).find('button');
        myButton.simulate("click");
        expect(onClickFunc.mock.calls.length).toEqual(1);
    })
})