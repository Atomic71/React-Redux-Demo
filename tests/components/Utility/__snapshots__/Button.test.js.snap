// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button utility component matches the expected snapshot 1`] = `
<button
  className="Button something"
  disabled={undefined}
  onClick={[MockFunction]}
  type={undefined}
>
  Andrej's Button
</button>
`;
